### PR TITLE
[WIP] Update suite to run against the latest

### PR DESCRIPTION
**Pull request by CaseyLeask**
There's a lot here I'm not clear on, like how to get Wattsi's full-page
version. After digging through
https://github.com/whatwg/html-build/blob/master/build.sh I have more of
an idea how this all sticks together, but there's so many different
toolchains it might take some time.

### CODE REVIEW

**Comment by benschwarz** [comment]
We might want to update anolis and spec splitter etc etc? I guess by running make you've already got recent versions?

**Comment by domenic** [comment]
So HTML doesn't use Anolis anymore. Wattsi is supposed to output a dev multipage version for you; you can see an older snapshot at https://html5.org/temp/multipage-dev.zip, or use Wattsi + build.sh to generate it directly.

**Comment by CaseyLeask** [comment]
Thanks for the fast response folks! Wasn't expecting anything back for a bit.

To borrow from @benschwarz, is this the updated todo list?
- [ ] Update the suite to run against the latest dev multipage edition https://html5.org/temp/multipage-dev.zip or use Wattsi + build.sh to generate it directly
- [ ]  Publish it to gh-pages
- [ ]  Point DNS at github (@hixie)
- [ ]  Update design, modernise front-end
- [ ]  Investigate more modern tooling (python spec-splitter update, remove janky ruby, etc) OR, make moves to make the role of this repository to style and deploy the specification (rather than the insane processing etc that it does now).

**Comment by CaseyLeask** [comment]
I've reached the point where nothing crashes and it emits a zero error code on `make`. From this point on, I'll try my best to match the current output at https://developers.whatwg.org/

**Comment by domenic** [comment]
That sounds about right, although some of the tooling might be obsolete, e.g. the spec comes pre-split. So maybe along the lines of your "OR".

On the other hand, if you get something working with updated content and auto-updating, that'll be a big improvement no matter the strategy :D

**Comment by benschwarz** [comment]
Sounds good @CaseyLeask — I want to see the spec updated with the latest generated copy, then move to work on improving the tooling / generation / deployment, and then finally start working on the design and functionality of it.

**Comment by benschwarz** [comment]
Thanks for the hard work so far @CaseyLeask :whale:

File: Rakefile [hunk 1]
```diff
@@ -71,19 +65,19 @@ namespace :postprocess do
...
-          item.remove
-        end
-      end
-    end
+    #    if a_href == item_href
+    #      item.remove
+    #    end
+    #  end
+    #end
   end
```
Comment (benschwarz) [inline Rakefile]: Are we able to get these disabled lines figured out? Any questions?

Comment (CaseyLeask) [inline Rakefile]: Currently it does this to the table of contents.

screen shot 2015-12-31 at 4 14 09 pm

Need to investigate more.

**Comment by CaseyLeask** [comment]
`head` is looking much better now. Onto the `body` of the doc.

**Comment by benschwarz** [comment]
:tada:

**Comment by annevk** [comment]
@CaseyLeask let us know if you want us to generate something like  https://html5.org/temp/multipage-dev.zip automatically and ping your build script somehow. That might be easier than running a fork of the build script and such.

Filing an issue against https://github.com/whatwg/html-build once you're ready for that is probably easiest.

(Thanks @benschwarz for pointing me towards this thread!)

**Comment by CaseyLeask** [comment]
@annevk That sounds like a good idea. I'm using https://html5.org/temp/multipage-dev.zip in this version.

I think I've still got a long way to go with this one. `spec-splitter` did a lot more wrangling than I gave it credit for.

**Comment by annevk** [comment]
Note that we could also change https://github.com/whatwg/wattsi if the output for the developer edition is not great. Wattsi ultimately generates that so if there are things that are not usable we can simply change them.

**Comment by CaseyLeask** [comment]
I'm keeping track of the issues in the branch, so I can keep a reasonable measure of progress.
https://github.com/CaseyLeask/developers.whatwg.org/issues

**Comment by annevk** [comment]
@CaseyLeask let us know if things would be better to update upstream. Since we already generate a multipage developer copy, tweaking that might be a better solution than post-processing. And ensure things stay synchronized.

**Comment by benschwarz** [comment]
@annevk, the biggest blocker is the lack of the TOC… a few other details, but that's the big one

**Comment by annevk** [comment]
The multipage-dev that whatwg/html-build generates and I posted an output copy from at https://html5.org/temp/multipage-dev.zip has both a TOC for all documents in `index.html` and each individual page also has a TOC at the beginning. I checked just now and this is also true if you generate a new more up-to-date copy from scratch.

**Comment by benschwarz** [comment]
@annevk Does it align with what's already deployed on developers.whatwg.org?

**Comment by annevk** [comment]
I think that is definitely the intent, yes. It's basically the HTML standard without the sections marked as being specific to implementers (which we might not always annotate carefully since there's no current output to inspect, but we should fix that in the source document). I don't know how you got there prior to this point, but this seems like a good strategy.

**Comment by benschwarz** [comment]
@CaseyLeask, I think we should merge what you have, then work on smaller specific issues from there, what's in master doesn't work at all, and this is at least half-way there. We can get it deploying to github pages etc, and start ironing out the kinks. 

Good with you?

**Comment by CaseyLeask** [comment]
@benschwarz Good with me.

**Comment by domenic** [comment]
Let us know when it's time to start setting up an automated system to deliver the dev version and deliver it to you.

**Pull request merged**